src/main/java/io/vertx/ext/mongo/impl/config/CompressorListParser.java

```diff
@@ -0,0 +1,77 @@
+package io.vertx.ext.mongo.impl.config;
+
+import com.mongodb.ConnectionString;
+import com.mongodb.MongoCompressor;
+import com.mongodb.lang.Nullable;
+import io.vertx.core.json.JsonObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CompressorListParser {
+
+  private final List<String> compressorKeys;
+  private final ConnectionString connectionString;
+  private final JsonObject config;
+
+  public CompressorListParser(ConnectionString connectionString, JsonObject config) {
+    this.connectionString = connectionString;
+    this.config = config;
+
+    this.compressorKeys = new ArrayList<>();
+    this.compressorKeys.add("compressors");
+    this.compressorKeys.add("zlibcompressionlevel");
+  }
+
+  public List<MongoCompressor> compressorList() {
+    String compressors = "";
+    Integer zlibCompressionLevel = null;
+
+    for (final String key : compressorKeys) {
+      String value = config.getString(key);
+      if (value == null) {
+        continue;
+      }
+
+      if (key.equals("compressors")) {
+        compressors = value;
+      } else if (key.equals("zlibcompressionlevel")) {
+        zlibCompressionLevel = Integer.parseInt(value);
```

Comment:
Please catch the `NumberFormatException` and rethrow as `IllegalArgumentException` for consistency with other parsers